docs/src/dev-guide/components-core/index.md

```diff
@@ -0,0 +1,97 @@
+# Core
+
+CLP core is the low-level component that performs compression, decompression, and search.
+
+## Requirements
+
+* We have built and tested CLP on the OSes listed [below](#native-environment).
+  * If you have trouble building for another OS, file an issue, and we may be able to help.
+* A compiler that supports C++17 and std::span (e.g., gcc-10)
```

Comment:
`log-surgeon` [requires](https://github.com/y-scope/log-surgeon/blob/main/src/log_surgeon/SchemaParser.cpp#L79) it. 

I agree it's odd, but I feel that's more the fault of the compilers since not even gcc-13 supports all of C++20, lol. So really this statement is saying that we need a compiler which supports all of C++17 plus one C++20 feature, `std::span`.